### DISCUSSION TITLE
what X library/interface does qtile use

### DISCUSSION THREAD

**Discussion by TH3-S4LM0N** [qtile/qtile#3760] [Q&A] [answered]
Ive been kinda switching between dwm and qtile for a while so i finally decided i would try making my own  x py wm using qtile as an example. My question is, how does qtile intrerface with the X server. Ive looked through all the import in `libqtile` but havent found anything.
basically
what x library should i use for this?

**Comment by elParaguayo** [comment, marked as answer]
We use `xcffib`.
